src/state_machine/coordinator/frost.rs

```diff
@@ -53,93 +53,61 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
         &mut self,
         packet: &Packet,
     ) -> Result<(Option<Packet>, Option<OperationResult>), Error> {
-        loop {
```

Comment:
You made a large change in this `fn` for `frost.rs` but not for `fire.rs`.  Why?

Reply:
Also, can you explain the reasoning for the changes?  You removed the `loop` which I thought was necessary for this `fn` to work properly

Reply:
The loop was required because it would switch to one of the states below which required it to do more work. It was always just a two step process. Now its just segregated into that two step process rather than the loop. And I am not touching FIRE until you are good with this change.

Reply:
So while the old code dispatched based on the current state and checked the message type, this code dispatches based on the message type and checks the state then calls the state dispatch function.  Then there are discrete additional steps for returning response packets, and another state based dispatch for ephemeral states, which also might return response packets.  And any message dispatch that can return an operation result gets a different return type which is checked.

I think I understand what is going on, but I'm still unsure as to why.  It's not obvious to me that this was necessary to accomplish the goals of this PR; what happened before you made this change?

Reply:
This change was necessary to determine what function to call based on what message is received. Before it looked at state and attempted to call a certain message type which of course failed because it would not match

Reply:
I might be able to rework it such that it only looks for specific coordinator "trigger" messages and then everything else remains the same, but I have my doubts that this will work.

Reply:
Nvm. You so smart. Done.